src/univariate/discrete/negativebinomial.jl

```diff
@@ -97,13 +97,15 @@ end
 
 # Implement native pdf and logpdf since it's relatively straight forward and allows for ForwardDiff
 function logpdf(d::NegativeBinomial, k::Real)
-    r = d.r * log(d.p) + k * log1p(-d.p)
-    if isone(d.p) && iszero(k)
-        return zero(r)
-    elseif !insupport(d, k)
-        return oftype(r, -Inf)
+    r, p = params(d)
+    z = xlogy(r, p) + xlog1py(k, -p)
+    if iszero(k)
+        # in this case `log(k + r) + logbeta(r, k + 1) == 0` analytically but unfortunately not numerically
+        return z
+    elseif insupport(d, k)
```

Comment:
I have a preference for early termination with edge cases and finishing with the standard case (as was done before, so a branch for !insupport and the normal computation otherwise